### PR TITLE
feat(key-management-service): list of kms implementation

### PR DESCRIPTION
**Pull request by vovh**
ref: MANAGER-13564

| Question         | Answer
| ---------------- | ---
| Branch?          | [feat/key-management-system](https://github.com/ovh/manager/tree/feat/key-management-system)
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #13564
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [x] Breaking change is mentioned in relevant commits

## Description

Currently, there is no Key Management Service implemented on the OVHcloud Manager.

A customer can’t create any key, credential or secret.

When a customer want to create a key, a credential or a secret, there is necessarily a wallet, called KMS, to stock all information.

The objective will be to propose this new feature, directly in the menu.

### CODE REVIEW

**Review by chipp972 — APPROVED** [review]

**Comment by chipp972** [comment]
You should rename your playwright tests to not be run by jest or vitest

File: packages/manager/apps/key-management-system/src/api/GET/apiv2/services.ts [hunk 1]
```diff
@@ -93,3 +96,49 @@ export const getListingIcebergV2 = async ({
...
+export const paginateResults = (items: OKMS[], pagination: PaginationState) => {
+  return {
+    rows: items.slice(
+      pagination.pageIndex * pagination.pageSize,
+      (pagination.pageIndex + 1) * pagination.pageSize,
+    ),
+    pageCount: Math.ceil(items.length / pagination.pageSize),
+    totalRows: items.length,
+  };
+};
```
Comment (darsene) [inline packages/manager/apps/key-management-system/src/api/GET/apiv2/services.ts:129]: You might don't need it as apiv2 still doesn't manage `pageIndex` and `totalCount`

File: packages/manager/apps/key-management-system/src/api/utils.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { OKMS } from '@/interface';
+
+export const defaultCompareFunction = (key: keyof OKMS) => (
```
Comment (darsene) [inline packages/manager/apps/key-management-system/src/api/utils.ts:3]: Do you really need to create a generic compare function whereas you only use it with `id` key ? Moreover, it won't work with `iam` key which is not a `string`

File: packages/manager/apps/key-management-system/src/api/GET/apiv2/services.ts [hunk 3]
```diff
@@ -93,3 +96,49 @@ export const getListingIcebergV2 = async ({
...
+  sorting: ColumnSort;
+};
+
+/**
+ *  Get okms listing with iceberg V2
+ */
+
+export const getListingIceberg = async () => {
+  try {
+    const List = await fetchIcebergV2({
```
Comment (darsene) [inline packages/manager/apps/key-management-system/src/api/GET/apiv2/services.ts:111]:
```suggestion
    const list = await fetchIcebergV2({
```

File: packages/manager-components/src/components/datagrid/region-cell.component.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  'AP_SOUTH_MUM' = 'Asia Pacific (India – Mumbai)',
+  'EU_WEST_GRA_SNC' = 'Europe (France - Gravelines) SNC ',
+  'EU_WEST_RBX_SNC' = 'Europe (France - Roubaix) SNC',
+  'EU_WEST_SBG_SNC' = 'Europe (France - Strasbourg) SNC',
+  'LABEU_WEST_1_PREPROD' = 'LABEU (France - Croix) (PREPROD)',
+  'LABEU_WEST_1_DEV_1' = 'LABEU (France - Croix) (DEV-1)',
+  'LABEU_WEST_1_DEV_2' = 'LABEU (France - Croix) (DEV-2)',
+  'EU_WEST_RBX_INT' = 'Europe (France - Roubaix) (Int)',
+  'EU_WEST_HDF_BACKUP' = 'Europe (France)',
+}
```
Comment (darsene) [inline packages/manager-components/src/components/datagrid/region-cell.component.tsx:27]: Why not managing translations for region name ? Has it been decided like that ?
Usually, we translate countries and cities in local language.
Example : https://github.com/ovh/manager/tree/master/packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/clusters/cluster/dashboard/general-information/translations

**Pull request closed without merging**